### PR TITLE
Connect Events on Homepage

### LINKED ISSUE #61: Connect Events and Announcements on Landing to Backend
[Project/Issue Tracker](https://github.com/Code-4-Community/lucys-love-bus-frontend/projects/1)

### Objective

We are using dummy data for the events and announcements on the homepage. 

As of the time you work on this ticket, events and announcements will be in our redux store. You'll have to add a dispatch to fetch them, and use them from the store. 
 
### Tech Spec

- [ ] Add a dispatch of the existing thunks to fetch events and announcements to the homepage
- [ ] Access the data from the store and provide it to the page's components.

### PR DESCRIPTION
**Pull request by rymaju**
## Issue

Closes #61 

## Changes

- Changed the homepage to take in upcoming events, and dispatch a request to fetch them,
-  I also edited the events card because the CSS was a little outdated

## Screenshots

Provide screenshots of any new components, styling changes, or pages. 
image

## Verification Steps
I added 4 events, only 3 show up on the homepage

### CODE REVIEW

**Comment by rymaju** [comment]
Not the fault of this PR that its failing, master was failing when pulled

File: src/components/EventCard.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ const DateText = styled(Text)`
 `;
 
 interface EventsCardProps {
-  src: string;
+  src?: string;
```
Comment (anirudh723) [inline src/components/EventCard.tsx:32]: Consider renaming this to `imageSrc` to be more descriptive, and consistent with naming conventions in the Announcement component.

File: src/components/EventCard.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const EventsCardComponent: React.FC<EventsCardProps> = ({
   to,
 }) => {
   return (
-    <EventsCard cover={<img alt="example" src={src} />}>
+    <EventsCard cover={<img alt="example" src={src || DEFAULT_IMAGE} />}>
```
Comment (anirudh723) [inline src/components/EventCard.tsx:47]: Do wanna apply default images only on the frontend, or should we create announcements with a default image on the backend if the imageSrc wasn't specified in the POST request?

In another PR review of your branch, I think I mentioned that the backend could do it, but I think I would be better to not manipulate the announcement data, and keep the default image application as a UI feature only. 

Wonder what your thoughts are @jackblanc

File: src/containers/home/index.tsx [hunk 3]
```diff
@@ -13,8 +14,10 @@ import {
...
 } from '../../utils/asyncRequest';
 import { ORANGE } from '../../utils/colors';
-import { AnnouncementsDataProps } from '../announcements';
 import { getAnnouncements } from '../announcements/ducks/thunks';
+import { AnnouncementsReducerState } from '../announcements/ducks/types';
+import { getUpcomingEvents } from '../upcoming-events/ducks/thunks';
+import { EventProps, EventsReducerState } from '../upcoming-events/ducks/types';
 const { Text, Paragraph } = Typography;
 const image1v2 =
   'https://lucys-love-bus-public.s3.us-east-2.amazonaws.com/sajni+center+thiago+music(1).jpg';
```
Comment (anirudh723) [inline src/containers/home/index.tsx:23]: Consider moving this to where we keep our other urls

File: src/containers/home/index.tsx [hunk 4]
```diff
@@ -118,38 +125,25 @@ const Home: React.FC<HomeContainerProps> = ({ announcements }) => {
...
-              src="https://americanhumane.org/app/uploads/2016/08/animals-cats-cute-45170-min.jpg"
-              title="VIRTUAL Slow Flow Restorative Yoga"
-              date={new Date()}
-              description={`Find some time for self-care and join yoga teacher Sarah Oleson
-              for a peaceful and rejuvenating virtual...`}
-              to="/"
-            />
-          </Col>
+          {asyncRequestIsFailed(events) && (
+            <p>The announcements could not be retrieved.</p>
```
Comment (anirudh723) [inline src/containers/home/index.tsx:129]: events*

File: src/containers/home/index.tsx [hunk 5]
```diff
@@ -118,38 +125,25 @@ const Home: React.FC<HomeContainerProps> = ({ announcements }) => {
...
+          )}
+          {asyncRequestIsLoading(announcements) && <p>Loading events...</p>}
```
Comment (anirudh723) [inline src/containers/home/index.tsx:131]: `asyncRequestIsLoading(events)`*

**Review by anirudh723 — APPROVED** [review]
Looks good to me! Left some quick-fix comments and also a question regarding the default images

File: src/containers/home/index.tsx (hunk 4 above)
Comment (rymaju) [reply src/containers/home/index.tsx:129]: good catch!

File: src/containers/home/index.tsx (hunk 5 above)
Comment (rymaju) [reply src/containers/home/index.tsx:131]: Oo yeah this was sloppy, this should be fixed

File: src/components/EventCard.tsx (hunk 2 above)
Comment (rymaju) [reply src/components/EventCard.tsx:47]: IMO its better to have this on the frontend, but I'll refrain from merging until we have a consensus

File: src/components/EventCard.tsx (hunk 1 above)
Comment (jackblanc) [reply src/components/EventCard.tsx:32]: agree

File: src/components/EventCard.tsx (hunk 2 above)
Comment (jackblanc) [reply src/components/EventCard.tsx:47]: I think the default image should be client-side, as long as we always intend for it to be the same image default for all occurrences. Makes it easier to change I suppose.

File: src/components/event-listing/EventListing.tsx [hunk 6]
```diff
@@ -48,13 +49,10 @@ const EventListing: React.FC<EventProps> = ({
...
   details,
 }) => {
-  const defaultImg =
-    'https://today.tamu.edu/wp-content/uploads/2019/03/GettyImages-184621282.jpg';
-
   return (
     <StyledCard>
       <CardContent>
-        <Thumbnail src={thumbnail || defaultImg} />
+        <Thumbnail src={thumbnail || DEFAULT_IMAGE} />
```
Comment (jackblanc) [inline src/components/event-listing/EventListing.tsx:55]: Another related thought: could you make `DEFAULT_IMAGE` a default function parameter? 
`thumbnail = DEFAULT_IMAGE` in function params? Not sure if that's proper syntax.

File: src/containers/home/index.tsx [hunk 7]
```diff
@@ -72,15 +74,19 @@ const ViewMoreButton = styled(LinkButton)`
   margin: 1em;
 `;
 
-const ANNOUNCEMENTS_LIMIT = 3;
+const CARD_ROW_LIMIT = 3;
 
-export type HomeContainerProps = AnnouncementsDataProps;
+export type HomeContainerProps = {
```
Comment (jackblanc) [inline src/containers/home/index.tsx:79]: type vs interface? why type?

**Review by jackblanc — APPROVED** [review]
Love it, left some thoughts about longer-term things

File: src/containers/home/index.tsx (hunk 7 above)
Comment (rymaju) [reply src/containers/home/index.tsx:79]:
```
type name = {
 ...
}
```
is the same as
```
interface name {
 ...
}
```

Comment (rymaju) [reply src/containers/home/index.tsx:79]: We should definitely do interface for consistency though 👍

File: src/components/event-listing/EventListing.tsx (hunk 6 above)
Comment (rymaju) [reply src/components/event-listing/EventListing.tsx]: I tried to do this in the prop deconstruction but it didnt work :/ I'm gonna gonna do it with ||

**Pull request merged**